src/send.c

```diff
@@ -111,25 +111,7 @@ void fastd_send(
 	if (!msg.msg_controllen)
 		msg.msg_control = NULL;
 
-	int ret = sendmsg(sock->fd.fd, &msg, 0);
-
-	if (ret < 0 && msg.msg_controllen) {
-		switch (errno) {
-		case EINVAL:
-		case ENETUNREACH:
-			pr_debug2("sendmsg: %s (trying again without pktinfo)", strerror(errno));
-
-			if (peer && !fastd_peer_handshake_scheduled(peer))
-				fastd_peer_schedule_handshake_default(peer);
-
-			msg.msg_control = NULL;
-			msg.msg_controllen = 0;
-
-			ret = sendmsg(sock->fd.fd, &msg, 0);
-		}
-	}
-
-	if (ret < 0) {
```

Comment:
Removing this code block breaks fast source address switching without a full socket timeout and reestablish.